src/sql/workbench/browser/modal/optionsDialog.ts

```diff
@@ -204,13 +206,79 @@ export class OptionsDialog extends Modal {
 	}
 
+	/**
+	 * Registers on selection change event for connection options configured with 'onSelectionChange' property.
+	 */
+	private registerOnSelectionChangeEvents(options: { [name: string]: any }, container: HTMLElement): void {
+		//Register on selection change event for all advanced options
+		for (let optionName in this._optionElements) {
+			let widget: Widget = this._optionElements[optionName].optionWidget;
+			if (widget instanceof SelectBox) {
+				this._registerSelectionChangeEvents([this._optionElements], this._optionElements[optionName].option, widget, container);
+			}
+		}
+	}
+
+	private _registerSelectionChangeEvents(collections: { [optionName: string]: OptionsDialogHelper.IOptionElement }[], option: azdata.ServiceOption, widget: SelectBox, container: HTMLElement) {
+		if (option?.onSelectionChange) {
+			option.onSelectionChange.forEach((event) => {
+				this._register(widget.onDidSelect(value => {
+					let selectedValue = value.selected;
+					event?.dependentOptionActions?.forEach((optionAction) => {
+						let defaultValue: string | undefined = collections[optionAction.optionName]?.option.defaultValue ?? '';
+						let widget: AdsWidget | undefined = this._findWidget(collections, optionAction.optionName);
+						if (widget) {
+							createCSSRule(`.hide-${widget.id} .option-${widget.id}`, `display: none;`);
+							this._onValueChangeEvent(container, selectedValue, event.values, widget, defaultValue, optionAction.action);
+						}
+					});
+				}));
+			});
+			// Clear selection change actions once event is registered.
+			option.onSelectionChange = undefined;
+			if (this.optionValues[option.name]) {
+				widget.selectWithOptionName(this.optionValues[option.name], true);
+			} else {
+				widget.select(0, true);
+			}
+		}
+	}
+
+	private _onValueChangeEvent(container: HTMLElement, selectedValue: string, acceptedValues: string[],
+		widget: AdsWidget, defaultValue: string, action: string): void {
+		if ((acceptedValues.includes(selectedValue.toLocaleLowerCase()) && action === Actions.Show)
+			|| (!acceptedValues.includes(selectedValue.toLocaleLowerCase()) && action === Actions.Hide)) {
+			container.classList.remove(`hide-${widget.id}`);
+		} else {
+			// Support more Widget classes here as needed.
+			if (widget instanceof SelectBox) {
+				widget.select(widget.values.indexOf(defaultValue));
+			} else if (widget instanceof InputBox) {
+				widget.value = defaultValue;
+			}
+			container.classList.add(`hide-${widget.id}`);
+			widget.hideMessage();
+		}
+	}
+
+	/**
+	 * Finds Widget from provided collection of widgets using option name.
+	 * @param collections collections of widgets to search for the widget with the widget Id
+	 * @param id Widget Id
+	 * @returns Widget if found, undefined otherwise
+	 */
+	private _findWidget(collections: { [optionName: string]: OptionsDialogHelper.IOptionElement }[], id: string): AdsWidget | undefined {
+		return collections.find(collection => collection[id].optionWidget);
```

Comment:
Oops, you'll need this instead
```suggestion
		return collections.find(collection => !!collection[id].optionWidget)?.optionWidget;
```

Reply:
It needs another [id] lookup - I got it compiled, will check-in after testing